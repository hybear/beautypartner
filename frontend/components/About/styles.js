import styled from 'styled-components';

export const Content = styled.main`
  display: grid;
  grid-template:
    'profile artwork' auto
    'about artwork' 10vh
    'design development' 60vh / 60% 40%;
  padding: 40px;
`;

export const Profile = styled.div`
  grid-area: profile;
  display: grid;
  align-items: center;
  grid-template-columns: 25% 75%;
  border-bottom: solid 1px ${({ theme }) => theme.color.lightGrey};
  padding-bottom: 40px;
  .picture {
    margin: 0 auto;
    max-height: 180px;
    max-width: 180px;
    border-radius: 100px;
    object-fit: cover;
  }
  .content {
    margin-left: 20px;
    &__sub {
      font-size: 1em;
      i {
        color: ${({ theme }) => theme.color.primary};
      }
    }
    &__name {
      font-size: 1.875em;
      color: ${({ theme }) => theme.color.primary};
      font-family: 'QuadraSans-Bold', sans-serif;
      margin: 5px 0 10px;
      small {
        font-size: 0.5em;
        font-family: 'QuadraSans-Regular', sans-serif;
      }
    }
    &__shortinfo {
      font-size: 1em;
    }
    &__social {
    }
  }
`;

export const Artwork = styled.div``;
export const About = styled.div``;
export const Design = styled.div``;
export const Development = styled.div``;
