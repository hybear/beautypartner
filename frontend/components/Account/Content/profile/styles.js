import styled from 'styled-components';
import { Text } from '../../../General';
import { Button as GButton } from '../../../General/Form';

export const ProfileContainer = styled.div`
  align-items: start;
  display: grid;
  grid-template:
    'title title' auto
    'profile password' auto
    / 60% 1fr;
  grid-gap: 0 20px;
`;

export const Profile = styled.div`
  background: ${({ theme }) => theme.color.lightGrey};
  border-radius: 0.2rem;
  color: ${({ theme }) => theme.color.grey};
  box-shadow: inset 0px 1px 5px 0px rgba(0, 0, 0, 0.15);
  display: grid;
  grid-area: profile;
  ${props => !props.handleProfile && `grid-template-columns: repeat(auto-fill, minmax(calc(100% / 3), 1fr))`};
  grid-gap: 5px;
  /* margin: 10px 0; */
  padding: 20px;
  position: relative;
  p {
    font-family: 'QuadraSans-Bold';
  }
`;

export const ProfileItem = styled.span`
  margin: 5px;
  p {
    margin-bottom: 5px;
    font-size: 1.2em;
  }
`;

export const Password = styled.div`
  form {
    align-items: start;
    display: grid;
    grid-template: 'email email' auto;
    grid-gap: 10px;
  }
`;

export const EmailReset = styled(Text)`
  grid-area: email;
  color: ${({ theme, success }) => success && theme.color.success};
  b {
    font-family: 'QuadraSans-Bold';
  }
`;

export const PasswordContainer = styled.div`
  grid-area: password;
  max-width: 30rem;
  padding: 20px;
  /* margin: 10px 0; */
  border-radius: 0.2rem;
  background: ${({ theme }) => theme.color.lightGrey};
  box-shadow: inset 0px 1px 5px 0px rgba(0, 0, 0, 0.15);
  color: ${({ theme }) => theme.color.grey};
  position: relative;
  > p {
    font-family: 'QuadraSans-Bold';
    font-size: 1.2em;
  }
`;

export const Button = styled(GButton)`
  ${({ profile }) => profile && `margin: 20px 10px 0 0;`}
`;
