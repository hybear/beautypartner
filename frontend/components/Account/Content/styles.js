import styled from 'styled-components';
import { Title as GTitle } from '../../General';

export const Content = styled.div`
  grid-area: cont;
  margin: 20px 0 0 40px;
  padding: 0 20px 0;
  overflow: auto;
  &::-webkit-scrollbar-track {
    box-shadow: inset 0 0 1px rgba(0, 0, 0, 0.3);
    border-radius: 10px;
    background-color: ${({ theme }) => theme.color.white};
  }

  &::-webkit-scrollbar {
    width: 0.5em;
    background-color: ${({ theme }) => theme.color.white};
  }

  &::-webkit-scrollbar-thumb {
    border-radius: 10px;
    background-color: ${({ theme }) => theme.color.lightGrey};
  }

  ${({ theme }) => theme.media.tablet`
      margin: 20px 0 80px;
  `}
`;

export const Title = styled(GTitle)`
  text-align: initial;
  font-size: 1.875em;
  font-family: 'QuadraSans-Bold';
  color: ${({ theme }) => theme.color.grey};
  margin-bottom: 20px;
`;
