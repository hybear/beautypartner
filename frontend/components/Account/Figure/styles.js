import styled, { keyframes } from 'styled-components';

export const FigureContainer = styled.div`
  grid-area: figures;
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100%;
  width: 100%;
  ${({ theme }) => theme.media.tablet`
    display: none;
  `}
`;

const bright = keyframes`
  0% {
    filter: contrast(80%);
  }

  50% {
    filter: contrast(120%);
  }

  100% {
    filter: contrast(80%);
  }
`;

export const Image = styled.img`
  mask-image: url('/assets/icon/MaskAccount.svg');
  mask-position: right;
  mask-repeat: no-repeat;
  mask-size: cover;
  object-fit: cover;
  object-position: right;
  height: 100%;
  width: 100%;
  animation: ${bright} 3s linear infinite;
`;
