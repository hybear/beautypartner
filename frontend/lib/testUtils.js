import casual from 'casual';

// seed it so we get consistent results
casual.seed(777);

const fakeItem = () => ({
  __typename: 'Item',
  id: 'abc123',
  listPrice: 180000,
  bestPrice: 155200,
  category: 100,
  reference: '15523',
  title: 'LA MER The Moisturizing Soft Cream',
  description:
    'An ultra-soft face moisturizer that nurtures for beautiful, glowing radiance, leaving skin looking naturally brightened, hydrated, and restored.',
  image: 'laMer.jpg',
  largeImage: 'largelaMer.jpg',
});

const fakeUser = () => ({
  __typename: 'User',
  id: '4234',
  name: casual.name,
  email: casual.email,
  permissions: ['ADMIN'],
  birthday: '1996-03-08T14:18:00.000Z',
  creditCards: [],
  phone: casual.phone,
  balance: 1800,
  orders: [],
  cart: [],
});

const fakeOrderItem = () => ({
  __typename: 'OrderItem',
  id: casual.uuid,
  image: `${casual.word}.jpg`,
  title: casual.words(),
  listPrice: 4234,
  bestPrice: 4234,
  quantity: 1,
  description: casual.words(),
});

const fakeOrder = () => ({
  __typename: 'Order',
  id: 'ord123',
  charge: 'ch_123',
  total: 400,
  cashback: 5,
  paymentMethod: 'stripe',
  items: [fakeOrderItem(), fakeOrderItem()],
  status: ['APROVED'],
  createdAt: '2020-02-17T01:52:31.269Z',
  user: fakeUser(),
});

const fakeCartItem = overrides => ({
  __typename: 'CartItem',
  id: 'omg123',
  quantity: 3,
  item: fakeItem(),
  user: fakeUser(),
  ...overrides,
});

// Fake LocalStorage
class LocalStorageMock {
  constructor() {
    this.store = {};
  }

  clear() {
    this.store = {};
  }

  getItem(key) {
    return this.store[key] || null;
  }

  setItem(key, value) {
    this.store[key] = value.toString();
  }

  removeItem(key) {
    delete this.store[key];
  }
}

export { LocalStorageMock, fakeItem, fakeUser, fakeCartItem, fakeOrder, fakeOrderItem };
