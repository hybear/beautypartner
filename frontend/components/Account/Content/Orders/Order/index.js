import PropTypes from 'prop-types';
import { Query } from 'react-apollo';
import gql from 'graphql-tag';
import formatMoney from '../../../../../lib/formatMoney';
import { format, formatDistance, parseISO } from 'date-fns';
import Link from 'next/link';

// UI
import { Title } from '../../styles'; // Content
import { OrderStylesContainer, OrderStyles, OrderInfo, OrderItems, OrderButton } from './styles';

// ANIMATIONS
import Loading from '../../../../General/Animations/ContentLoading';

const SINGLE_ORDER_QUERY = gql`
  query SINGLE_ORDER_QUERY($id: ID!) {
    order(id: $id) {
      id
      charge
      total
      createdAt
      status
      cashback
      paymentMethod
      user {
        id
      }
      items {
        id
        title
        description
        image
        listPrice
        bestPrice
        quantity
      }
    }
  }
`;

const Order = ({ id }) => {
  return (
    <Query query={SINGLE_ORDER_QUERY} variables={{ id }}>
      {({ data, error, loading }) => {
        // if (error) return <Error error={error} />;
        if (loading) return <Loading isStoped={false} />;

        const { order } = data;
        console.log(order);

        return (
          <OrderStylesContainer>
            <OrderStyles>
              <Title>Order: {order.id}</Title>
              <OrderInfo>
                <span className="title">Status</span>
                <span className="title">Date</span>
                <span className="title">Total</span>
                <span className="title">Cashback</span>
                <span className="content status">{order.status}</span>
                <span className="content date">{formatDistance(parseISO(order.createdAt), new Date())} ago</span>
                <span className="content total">{formatMoney(order.total)}</span>
                <span className="content cashback">
                  +
                  {formatMoney(
                    (order.items.reduce((tally, cartItem) => tally + cartItem.bestPrice * cartItem.quantity, 0) *
                      order.cashback) /
                      100
                  )}
                  <small>{order.cashback}%</small>
                </span>
              </OrderInfo>
              <OrderItems>
                <span className="title">Photo</span>
                <span className="title">Title</span>
                <span className="title">Quantity</span>
                <span className="title">Unit Price</span>
                <span className="title">Total</span>
                {order.items.map(item => (
                  <div className="item">
                    <img className="item__picture" src={item.image} alt={item.title} />
                    <p>{item.title}</p>
                    <p>{item.quantity}</p>
                    <p>{formatMoney(item.bestPrice)}</p>
                    <p>{formatMoney(item.bestPrice * item.quantity)}</p>
                  </div>
                ))}
              </OrderItems>
            </OrderStyles>
          </OrderStylesContainer>
        );
      }}
    </Query>
  );
};

Order.PropTypes = {
  id: PropTypes.string.isRequired,
};

export default Order;
