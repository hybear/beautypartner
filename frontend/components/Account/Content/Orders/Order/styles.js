import styled from 'styled-components';
import { Title } from '../../styles';
import { Button } from '../../../../General/Form';

export const OrderStylesContainer = styled.div`
  height: 100%;
  overflow: hidden;
`;

export const OrderTotal = styled.p`
  font-size: 1.1em;
  margin: 0 0 15px;
`;
export const OrderListTitle = styled.div`
  /* display: grid;
  grid-template-columns: repeat(5, auto);
  text-align: center; */
`;
export const OrderStyles = styled.li`
  margin: 10px 0;
  display: block;
  .order {
    display: grid;
    justify-content: space-between;
    align-items: center;
    grid-template-columns: repeat(4, auto);
    padding: 20px;
    transition: all 0.2s;
    text-align: center;
    color: ${({ theme }) => theme.color.grey};
    text-decoration: none;
    background: ${({ theme }) => theme.color.lightGrey};
    box-shadow: inset 0px 1px 5px 0px rgba(0, 0, 0, 0.15);
    border-radius: 0.2rem;

    .order__title {
      font-family: 'QuadraSans-Bold', sans-serif;
      margin-bottom: 10px;
      font-size: 1.25em;
    }

    .order__info {
      &.cashback {
        small {
          display: block;
          background: ${({ theme }) => theme.color.primary};
          color: ${({ theme }) => theme.color.white};
          padding: 0.1em 0;
          border-radius: 2px;
        }
      }
    }
  }
`;

export const OrderButton = styled(Button)`
  a {
    color: ${({ theme }) => theme.color.white};
    &:hover {
      text-decoration: none;
    }
  }
`;

export const OrderList = styled.ul`
  overflow-x: hidden;
  overflow-y: auto;
  margin-bottom: 20px;
  &::-webkit-scrollbar-track {
    box-shadow: inset 0 0 1px rgba(0, 0, 0, 0.3);
    border-radius: 10px;
    background-color: #f5f5f5;
  }

  &::-webkit-scrollbar {
    width: 0.5em;
    background-color: #f5f5f5;
  }

  &::-webkit-scrollbar-thumb {
    border-radius: 10px;
    background-color: ${({ theme }) => theme.grey};
  }
`;
