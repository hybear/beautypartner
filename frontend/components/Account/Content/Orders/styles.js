import styled from 'styled-components';
import { Title } from '../styles';

export const OrderStylesContainer = styled.div`
  height: 100%;
  overflow: hidden;
`;

export const OrderTotal = styled.p`
  font-size: 1.1em;
  margin: 0 15px 20px;
`;

export const OrderList = styled.ul`
  height: 100%;
  overflow-x: hidden;
  overflow-y: auto;
  padding-bottom: 40px;
  &::-webkit-scrollbar-track {
    box-shadow: inset 0 0 1px rgba(0, 0, 0, 0.3);
    border-radius: 10px;
    background-color: #f5f5f5;
  }

  &::-webkit-scrollbar {
    width: 0.5em;
    background-color: #f5f5f5;
  }

  &::-webkit-scrollbar-thumb {
    border-radius: 10px;
    background-color: ${({ theme }) => theme.grey};
  }
`;
