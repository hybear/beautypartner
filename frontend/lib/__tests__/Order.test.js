import Order, { SINGLE_ORDER_QUERY } from '@components/Account/Content/Orders/Order';
import { MockedProvider } from '@apollo/react-testing';
import { act } from 'react-dom/test-utils';
import wait from 'waait';
import { fakeOrder } from '../testUtils';
import toJson from 'enzyme-to-json';
import { ThemeProviderWrapper } from '../../styles/theme';

const mocks = [
  {
    request: { query: SINGLE_ORDER_QUERY, variables: { id: 'abc123' } },
    result: {
      data: {
        order: fakeOrder(),
      },
    },
  },
];

describe('<Order/>', () => {
  it('renders and displays information properly', async () => {
    await act(async () => {
      const wrapper = mount(
        <MockedProvider mocks={mocks}>
          <Order id="abc123" />
        </MockedProvider>,
        {
          wrappingComponent: ThemeProviderWrapper,
        }
      );

      await wait(0);
      wrapper.update();

      expect(toJson(wrapper.find('.totalVal')).children[1]).toContain('400.00');
      expect(toJson(wrapper.find('.cashbackVal')).children[1]).toBe('$4.23');
    });
  });

  it('matches snapshot', async () => {
    await act(async () => {
      const wrapper = mount(
        <MockedProvider mocks={mocks}>
          <Order id="abc123" />
        </MockedProvider>,
        {
          wrappingComponent: ThemeProviderWrapper,
        }
      );

      await wait(0);
      wrapper.update();

      expect(wrapper.find('[data-test="order"]')).toMatchSnapshot();
    });
  });
});
