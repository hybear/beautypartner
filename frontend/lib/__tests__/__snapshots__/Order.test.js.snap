// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Order/> matches snapshot 1`] = `
Array [
  <styles__OrderStyles
    data-test="order"
  >
    <li
      className="styles__OrderStyles-n7vcma-3 fzqfIU"
      data-test="order"
    >
      <styles__OrderTitle>
        <h2
          className="Title-qmiaj1-0 styles__Title-sc-1ht2lvb-1 styles__OrderTitle-n7vcma-2 pjoeL"
        >
          Order: 
          ord123
        </h2>
      </styles__OrderTitle>
      <styles__OrderInfo>
        <div
          className="styles__OrderInfo-n7vcma-4 YySyk"
        >
          <span
            className="title status"
          >
            Status
          </span>
          <span
            className="title date"
          >
            Date
          </span>
          <span
            className="title total"
          >
            Total
          </span>
          <span
            className="title cashback"
          >
            Cashback
          </span>
          <span
            className="content statusVal"
          >
            APROVED
          </span>
          <span
            className="content dateVal"
          >
            10 days
             ago
          </span>
          <span
            className="content totalVal"
          >
            $400
          </span>
          <span
            className="content cashbackVal"
          >
            +
            $4.23
            <small>
              5
              %
            </small>
          </span>
        </div>
      </styles__OrderInfo>
      <styles__OrderItems>
        <div
          className="styles__OrderItems-n7vcma-5 jTvrcM"
        >
          <span
            className="title"
          >
            Photo
          </span>
          <span
            className="title"
          >
            Title
          </span>
          <span
            className="title"
          >
            Quantity
          </span>
          <span
            className="title"
          >
            Unit Price
          </span>
          <span
            className="title"
          >
            Total
          </span>
          <div
            className="item"
            key="0"
          >
            <img
              alt="soluta non omnis consequatur enim quia autem"
              className="item__picture"
              src="reprehenderit.jpg"
            />
            <p>
              soluta non omnis consequatur enim quia autem
            </p>
            <p>
              1
            </p>
            <p>
              $42.34
            </p>
            <p>
              $42.34
            </p>
          </div>
          <div
            className="item"
            key="1"
          >
            <img
              alt="quia sed exercitationem omnis laborum exercitationem est"
              className="item__picture"
              src="sint.jpg"
            />
            <p>
              quia sed exercitationem omnis laborum exercitationem est
            </p>
            <p>
              1
            </p>
            <p>
              $42.34
            </p>
            <p>
              $42.34
            </p>
          </div>
        </div>
      </styles__OrderItems>
    </li>
  </styles__OrderStyles>,
  <li
    className="styles__OrderStyles-n7vcma-3 fzqfIU"
    data-test="order"
  >
    <styles__OrderTitle>
      <h2
        className="Title-qmiaj1-0 styles__Title-sc-1ht2lvb-1 styles__OrderTitle-n7vcma-2 pjoeL"
      >
        Order: 
        ord123
      </h2>
    </styles__OrderTitle>
    <styles__OrderInfo>
      <div
        className="styles__OrderInfo-n7vcma-4 YySyk"
      >
        <span
          className="title status"
        >
          Status
        </span>
        <span
          className="title date"
        >
          Date
        </span>
        <span
          className="title total"
        >
          Total
        </span>
        <span
          className="title cashback"
        >
          Cashback
        </span>
        <span
          className="content statusVal"
        >
          APROVED
        </span>
        <span
          className="content dateVal"
        >
          10 days
           ago
        </span>
        <span
          className="content totalVal"
        >
          $400
        </span>
        <span
          className="content cashbackVal"
        >
          +
          $4.23
          <small>
            5
            %
          </small>
        </span>
      </div>
    </styles__OrderInfo>
    <styles__OrderItems>
      <div
        className="styles__OrderItems-n7vcma-5 jTvrcM"
      >
        <span
          className="title"
        >
          Photo
        </span>
        <span
          className="title"
        >
          Title
        </span>
        <span
          className="title"
        >
          Quantity
        </span>
        <span
          className="title"
        >
          Unit Price
        </span>
        <span
          className="title"
        >
          Total
        </span>
        <div
          className="item"
          key="0"
        >
          <img
            alt="soluta non omnis consequatur enim quia autem"
            className="item__picture"
            src="reprehenderit.jpg"
          />
          <p>
            soluta non omnis consequatur enim quia autem
          </p>
          <p>
            1
          </p>
          <p>
            $42.34
          </p>
          <p>
            $42.34
          </p>
        </div>
        <div
          className="item"
          key="1"
        >
          <img
            alt="quia sed exercitationem omnis laborum exercitationem est"
            className="item__picture"
            src="sint.jpg"
          />
          <p>
            quia sed exercitationem omnis laborum exercitationem est
          </p>
          <p>
            1
          </p>
          <p>
            $42.34
          </p>
          <p>
            $42.34
          </p>
        </div>
      </div>
    </styles__OrderItems>
  </li>,
]
`;
